tests/Extensions/SearchFormFactoryExtensionTest.php

```diff
@@ -19,50 +18,24 @@ class SearchFormFactoryExtensionTest extends SapphireTest
         '../pages.yml',
     ];
 
-    public function testDefaultConfigValues(): void
-    {
-        $expected = ['SilverStripe\Assets\Image'];
-        $actual = Config::inst()->get(SearchFormFactoryExtension::class, 'exclude_classes');
-        $this->assertEquals($expected, $actual);
-    }
-
-    public function testImageAndFileInclusionInShowInSearch(): void
+    /**
+     * Ensure that the SearchIndexed field is added to the search forms for files and images
+     */
+    public function testUpdateForm(): void
     {
         $form = Form::create();
-        $fields = new FieldList(new TabSet('Editor'));
-        $form->setFields($fields);
+        $fieldsList = new FieldList(new TabSet('Editor'));
+        $form->setFields($fieldsList);
 
-        $image = $this->objFromFixture(Image::class, 'image');
-        // Every file has default ShowInSearch value of 1
-        // (https://github.com/silverstripe/silverstripe-assets/blob/2/src/File.php#L163)
-        $this->assertEquals(1, $image->ShowInSearch);
+        $fields = $form->Fields();
+        $this->assertNull($fields->fieldByName('SearchIndexed'));
 
+        $image = $this->objFromFixture(Image::class, 'image');
         $searchFormFactoryExtension = new SearchFormFactoryExtension();
         $searchFormFactoryExtension->updateForm($form, null, 'Form', ['Record' => $image]);
-        // By default, `SilverStripe\Assets\Image` is excluded from the search - see `_config/extensions.yml`
-        $this->assertEquals(0, $image->ShowInSearch);
-
-        $file = $this->objFromFixture(File::class, 'pdf-file');
-        $searchFormFactoryExtension->updateForm($form, null, 'Form', ['Record' => $file]);
-        $this->assertEquals(1, $file->ShowInSearch);
-    }
 
-    public function testExcludedFileExtensionShowInSearch(): void
-    {
-        // Modify config to exclude pdf files from search
-        Config::modify()->set(SearchFormFactoryExtension::class, 'exclude_file_extensions', ['pdf']);
-
-        $file = $this->objFromFixture(File::class, 'pdf-file');
-        // Default ShowInSearch value of 1
-        $this->assertEquals(1, $file->ShowInSearch);
-
-        $form = Form::create();
-        $fields = new FieldList(new TabSet('Editor'));
-        $form->setFields($fields);
-
-        $searchFormFactoryExtension = new SearchFormFactoryExtension();
-        $searchFormFactoryExtension->updateForm($form, null, 'Form', ['Record' => $file]);
-        $this->assertEquals(0, $file->ShowInSearch);
+        $fields = $form->Fields()->findOrMakeTab('Editor.Details');
+        $this->assertInstanceOf(DatetimeField::class, $fields->fieldByName('SearchIndexed'));
```

Comment:
could we have a case that also checks the canIndexInSearch works? were the warning field is displayed?